rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java

```diff
@@ -321,6 +350,35 @@ private void processBlockMessage(@Nonnull final MessageChannel sender, @Nonnull
         tryRelayBlock(sender, block, result);
         recordEvent(sender, EventType.VALID_BLOCK);
         Metrics.processBlockMessage("finish", block, sender.getPeerNodeID());
+
+        // If still synchronizing then return
+        if (blockProcessor.hasBetterBlockToSync()) {
+            return;
+        }
+
+        // If we are a valid source for federation notifications then generate one
```

Comment:
Why would this be here? Doesn't the federate already generate the notifications?